src/features/wallet/helpers/checkAvailableRewards.ts

```diff
@@ -0,0 +1,30 @@
+import { ethers } from "ethers";
+import { VoteRevealed } from "web3/get/queryVotesRevealedEvents";
+import { queryRetrieveRewards } from "web3/get/queryRetrieveRewards";
+
+export default async function checkAvailableRewards(
+  data: VoteRevealed[],
+  address: string,
+  contract: ethers.Contract
+) {
+  const promises = data.map(async (vote) => {
+    const rewardAvailable = await queryRetrieveRewards(
+      contract,
+      address,
+      vote.roundId,
+      vote.identifier,
+      vote.time
+    );
+    if (rewardAvailable) return parseFloat(rewardAvailable);
+  });
+
+  return Promise.all(promises).then((values) => {
+    let balance = 0;
+    values.map((val) => {
+      if (val && val) return (balance += val);
+      return false;
+    });
```

Comment:
im surprised typescript is ok with this, returning bool or number.  just a nit but id probably recommend doing this as a reduce operation
```js
const values = await Promise.all(promises)
return values.reduce((balance,val)=>{
  if (val && val) return balance + val;
  return balance
},0)

```

Reply:
I refactored it to this, a balance can be undefined (promise):

  return values.reduce((balance, val) => {
    if (val) return balance ? balance : 0 + val;
    return balance;
  }, 0);